### PR TITLE
Update prettybot workflow 

### PR DESCRIPTION
**Pull request by edlyn-liew-octo**
This PR fixed the prettybot workflow by adding the following. 

- Updated docker login to v3 
- Allow write access to github repo files to allow writeable permissions to all users (including the github actions runner ) to modify the files when the prettybot script is run  
- Updated git initialisation

### CODE REVIEW

File: .github/workflows/prettybot-automation.yml [hunk 1]
```diff
@@ -33,29 +33,32 @@ jobs:
...
                   else
                     echo "changed=false" >> $GITHUB_OUTPUT
                   fi
 
             - name: Commit and push changes
               if: steps.git-check.outputs.changed == 'true'
               run: |
-                  git commit -am "Run ReSharper code cleanup"
-                  git push origin ${{ env.BRANCH_NAME }}
+                git config user.email "bob@octopus.com"
```
Comment (ahaw2024) [inline .github/workflows/prettybot-automation.yml:58]: might we use builds@octopus.com here as bob@octopus.com is a fairly generic email

File: .github/workflows/prettybot-automation.yml [hunk 2]
```diff
@@ -33,29 +33,32 @@ jobs:
...
+                git config user.name "Github Actions"
```
Comment (ahaw2024) [inline .github/workflows/prettybot-automation.yml:59]: as above could we use something more meaningful here?

File: .github/workflows/prettybot-automation.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@ on:
     push:
         branches:
             - main
-
+
```
Comment (cailyoung) [inline .github/workflows/prettybot-automation.yml:7]:
```suggestion

```

File: .github/workflows/prettybot-automation.yml [hunk 4]
```diff
@@ -33,29 +33,32 @@ jobs:
...
+              uses: docker/login-action@v3
               with:
                   registry: ${{ secrets.INTERNAL_DOCKER_REGISTRY_URL }}
                   username: ${{ secrets.INTERNAL_DOCKER_READ_ONLY_USER_NAME }}
                   password: ${{ secrets.INTERNAL_DOCKER_READ_ONLY_ACCESS_TOKEN }}
-            
             - name: Run Cleanup Code
               run: |
-                docker run -v ${{ github.workspace }}:/src ${{ secrets.INTERNAL_DOCKER_REGISTRY_URL }}/octopusdeploy/tool-containers/tool-resharper-cli "/scripts/prettybot.sh"
+                chmod -R a+w ${{ github.workspace }}
```
Comment (cailyoung) [inline .github/workflows/prettybot-automation.yml:43]: Consider leaving a breadcrumb here as to _why_ we are `chmod`ing

File: .github/workflows/prettybot-automation.yml [hunk 5]
```diff
@@ -33,29 +33,32 @@ jobs:
...
+                docker run -v ${{ github.workspace }}:/src ${{ secrets.INTERNAL_DOCKER_REGISTRY_URL }}/octopusdeploy/tool-containers/tool-resharper-cli:latest /scripts/prettybot.sh
+            
             - name: Git Check for file changes
               id: git-check
               run: |
-                  if git status --porcelain ; then
-                    echo "changed=true" >> $GITHUB_OUTPUT
+                if git status --porcelain ; then
```
Comment (cailyoung) [inline .github/workflows/prettybot-automation.yml:49]: Intentional indentation change?

File: .github/workflows/prettybot-automation.yml (hunk 1 above)
Comment (cailyoung) [reply .github/workflows/prettybot-automation.yml:58]: I think we'd need to validate if we can merge a commit to this repo (or any other) with another committer identity, given the repo access controls. @edlyn-liew-octo do you mind closing #215 and trying again with a completely dummy git committer?

Comment (edlyn-liew-octo) [reply .github/workflows/prettybot-automation.yml:62]: Raised for a seperate conversation in [sc-81385]

**Review by cailyoung — APPROVED** [review]

**Pull request merged**